src/analysisd/decoders/winevtchannel.c

```diff
@@ -85,84 +149,40 @@ int DecodeWinevt(Eventinfo *lf){
     cJSON *json_system_in = cJSON_CreateObject();
     cJSON *json_eventdata_in = cJSON_CreateObject();
     cJSON *json_extra_in = cJSON_CreateObject();
+    cJSON *received_event = NULL;
     int level_n;
     unsigned long long int keywords_n;
     XML_NODE node, child;
-    int num;
     char *extra = NULL;
     char *filtered_string = NULL;
     char *level = NULL;
     char *keywords = NULL;
     char *msg_from_prov = NULL;
     char *returned_event = NULL;
     char *event = NULL;
-    char *find_event = NULL;
-    char *end_event = NULL;
-    char *real_end = NULL;
     char *find_msg = NULL;
-    char *end_msg = NULL;
-    char *next = NULL;
     char *severityValue = NULL;
     char *join_data = NULL;
     char *join_data2 = NULL;
-    char aux = 0;
     lf->decoder_info = winevt_decoder;
 
-    os_calloc(OS_MAXSTR, sizeof(char), event);
     os_calloc(OS_MAXSTR, sizeof(char), msg_from_prov);
     os_calloc(OS_MAXSTR, sizeof(char), join_data);
 
-    find_event = strstr(lf->log, "Event");
-
-    if(find_event){
-        find_event = find_event + 8;
-        end_event = strchr(find_event, '"');
-
-        if(end_event){
-            real_end = end_event;
-            aux = *(end_event + 1);
-
-            if(aux != '}' && aux != ','){
-                while(1){
-                    next = real_end + 1;
-                    real_end = strchr(next,'"');
-
-                    if(real_end) {
-                        aux = *(real_end + 1);
-                        if (aux == '}' || aux == ','){
-                            end_event = real_end;
-                            break;
-                        }
-                    } else {
-                        mdebug1("Malformed 'Event' field");
-                        break;
-                    }
-                }
-            }
-
-            num = end_event - find_event;
 
-            if(num > OS_MAXSTR - 1){
-                mwarn("The event message has exceeded the maximum size.");
-                cJSON_Delete(json_system_in);
-                cJSON_Delete(json_event);
-                cJSON_Delete(json_eventdata_in);
-                cJSON_Delete(json_extra_in);
-                ret_val = 1;
-                goto cleanup;
-            }
-
-            memcpy(event, find_event, num);
-            event[num] = '\0';
-            find_event = NULL;
-            end_event = NULL;
-            real_end = NULL;
-            next = NULL;
-        }
-    } else {
-        mdebug1("Malformed JSON output received. No 'Event' field found");
+    if (received_event = cJSON_Parse(lf->log), !received_event)
```

Comment:
We have to replace this by `cJSON_ParseWithOpts()` to be adapted to changes of https://github.com/wazuh/wazuh/pull/3648